azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -646,18 +646,11 @@ def compose_socket_options():
                 (socket.IPPROTO_TCP, socket.TCP_KEEPINTVL, INTERVAL_SECONDS),
                 (socket.IPPROTO_TCP, socket.TCP_KEEPCNT, MAX_FAILED_KEEPALIVES),
             ]
-        if (
-            sys.platform == "win32"
-            and hasattr(socket, "SOL_SOCKET")
-            and hasattr(socket, "SO_KEEPALIVE")
-            and hasattr(socket, "TCP_KEEPIDLE")
-            and hasattr(socket, "TCP_KEEPCNT")
-        ):
-            return [
-                (socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1),
-                (socket.IPPROTO_TCP, socket.TCP_KEEPIDLE, MAX_IDLE_SECONDS),
-                (socket.IPPROTO_TCP, socket.TCP_KEEPCNT, MAX_FAILED_KEEPALIVES),
-            ]
+        if sys.platform == "win32" and hasattr(socket, "SOL_SOCKET") and hasattr(socket, "SO_KEEPALIVE"):

```

Comment:
This is not a simple nit change.
What is your intention here?

Reply:
You're right, I should have documented it better.

When I installed python 3.5 to test it, it didn't compile anymore.
According to the docs:
`On Windows, TCP_KEEPIDLE, TCP_KEEPINTVL appear if run-time Windows supports.`
https://docs.python.org/3/library/socket.html

So since it's not guaranteed to be there, and didn't for me in practice I added this check.

Reply:
@yihezkel please review this change